include/circt/Dialect/SystemC/SystemCStatements.td

```diff
@@ -108,6 +108,32 @@ def InstanceDeclOp : SystemCOp<"instance.decl", [
   }];
 }
 
+def BindPortOp : SystemCOp<"instance.bind_port", [HasParent<"CtorOp">]> {
+  let summary = "Binds a port of a module instance to a channel.";
+  let description = [{
+    The ports of a submodule have to be bound to channels in a module further
+    up in the instance hierarchy (as opposed to `sc_export` where the channel
+    has to reside in the same module or a submodule). Therefore, a port can be
+    bound to either a signal declared by the `systemc.signal` operation or to
+    a port with matching direction (and thus bound to a channel further up in
+    the hierarchy).
+    More information on ports can be found in IEEE 1666-2011 §5.12.,
+    in particular IEEE 1666-2011 §5.12.7. is about port binding.
+    More information on predefined channels can be found in IEEE 1666-2011 §6.
+  }];
+
+  let arguments = (ins ModuleType:$instance,
+                       StrAttr:$portName,
+                       ChannelType:$channel);
+
+  let assemblyFormat = [{
+    $instance `[` $portName `]` `to` $channel attr-dict `:`
+    type($instance) `,` type($channel)
```

Comment:
It occurred to me that this type signature is somewhat verbose. It could be shortened to just be the type of the channel. However, that is not really correct. I couldn't come up with something less verbose that still made sense to me.

Reply:
I agree. The verbose part is the instance type because it includes all the ports, however, everything is needed for the op to work (except when you get rid of verifying that the port type is valid for the channel). Maybe a good way to reduce the verbosity of the IR (especially because there are usually a lot of `bind_port` ops operating on the same instance plus the `instance_decl` itself) would be type aliases for each module, e.g.,
```mlir
hw.type_scope @__systemc_typedecls {
  hw.typedecl @somemodule : !systemc.module<somemodule(in0: !systemc.in<i32>, out0: !systemc.out<i32>>
}
!somemodule = !hw.typealias<@__systemc_typedecls::@somemodule, !systemc.module<somemodule(in0: !systemc.in<i32>, out0: !systemc.out<i32>>>
systemc.module @somemodule (%in0: !systemc.in<i32>) -> (%out0: !systemc.out<i32>) {}
```
could reduce the verbosity from
```mlir
systemc.module @topmodule (%in0: !systemc.in<i32>) -> (%out0: !systemc.out<i32>) {
  %inst = systemc.instance.decl @somemodule : !systemc.module<somemodule(in0: !systemc.in<i32>, out0: !systemc.out<i32>>
  systemc.ctor {
    systemc.instance.bind_port %inst["in0"] to %in0 : !systemc.module<somemodule(in0: !systemc.in<i32>, out0: !systemc.out<i32>>, !systemc.in<i32>
    systemc.instance.bind_port %inst["out0"] to %out0 : !systemc.module<somemodule(in0: !systemc.in<i32>, out0: !systemc.out<i32>>, !systemc.out<i32>
  }
}
```
to
```mlir
systemc.module @topmodule (%in0: !systemc.in<i32>) -> (%out0: !systemc.out<i32>) {
  %inst = systemc.instance.decl @somemodule : !somemodule
  systemc.ctor {
    systemc.instance.bind_port %inst["in0"] to %in0 : !somemodule, !systemc.in<i32>
    systemc.instance.bind_port %inst["out0"] to %out0 : !somemodule, !systemc.out<i32>
  }
}
```
It adds the cost of looking up the type aliases, but when caching it that should be reasonably fast, right?

Reply:
Interesting idea. It is likely not worth it to add the alias just to reduce verbosity. However, this may be useful to revisit for other reasons in the future.

For the record, I'm totally happy with the current situation and the verbosity. No changes necessary. 👍